internal/kafka/internal/workers/cluster_mgrs/cleanup_clusters_mgr.go

```diff
@@ -0,0 +1,113 @@
+	"github.com/golang/glog"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	cleanupClustersWorkerType = "cleanup_clusters"
+)
+
+// CleanupClustersManager represents a worker that periodically reconciles data plane clusters in cleanup state.
+type CleanupClustersManager struct {
+	workers.BaseWorker
+	clusterService             services.ClusterService
+	kasFleetshardOperatorAddon services.KasFleetshardOperatorAddon
+	osdIdpKeycloakService      sso.OsdKeycloakService
+}
+
+// NewCleanupClustersManager creates a new worker that reconciles data plane clusters in cleanup state.
+func NewCleanupClustersManager(reconciler workers.Reconciler,
+	clusterService services.ClusterService,
+	kasFleetshardOperatorAddon services.KasFleetshardOperatorAddon,
+	osdIdpKeycloakService sso.OsdKeycloakService) *CleanupClustersManager {
+	return &CleanupClustersManager{
+		BaseWorker: workers.BaseWorker{
+			Id:         uuid.New().String(),
+			WorkerType: cleanupClustersWorkerType,
+			Reconciler: reconciler,
+		},
+		clusterService:             clusterService,
+		kasFleetshardOperatorAddon: kasFleetshardOperatorAddon,
+		osdIdpKeycloakService:      osdIdpKeycloakService,
+	}
+}
+
+// Start initializes the worker to reconcile data plane clusters in cleanup state.
+func (m *CleanupClustersManager) Start() {
+	m.StartWorker(m)
+}
+
+// Stop causes the process for reconciling data plane clusters in cleanup state to stop.
+func (m *CleanupClustersManager) Stop() {
+	m.StopWorker(m)
+}
+
+func (m *CleanupClustersManager) Reconcile() []error {
+	glog.Infoln("reconciling clusters")
+
+	var errList fleeterrors.ErrorList
+	err := m.processCleanupClusters()
+	if err != nil {
+		errList.AddErrors(err)
+	}
+
+	return errList.ToErrorSlice()
+}
+
+func (m *CleanupClustersManager) processCleanupClusters() error {
+	var errList fleeterrors.ErrorList
+
+	cleanupClusters, serviceErr := m.clusterService.ListByStatus(api.ClusterCleanup)
+	if serviceErr != nil {
+		errList.AddErrors(errors.Wrap(serviceErr, "failed to list of cleaup clusters"))
+		return errList
+	}
+
+	glog.Infof("cleanup clusters count = %d", len(cleanupClusters))
+
+	for _, cluster := range cleanupClusters {
+		glog.V(10).Infof("cleanup cluster ClusterID = %s", cluster.ClusterID)
+		metrics.UpdateClusterStatusSinceCreatedMetric(cluster, api.ClusterCleanup)
+		if err := m.reconcileCleanupCluster(cluster); err != nil {
+			errList.AddErrors(errors.Wrapf(err, "failed to reconcile cleanup cluster %s", cluster.ID))
+		}
+	}
+
+	if len(errList) > 0 {
+		return errList
+	}
+
+	return nil
+}
+
+func (m *CleanupClustersManager) reconcileCleanupCluster(cluster api.Cluster) error {
+	glog.Infof("Removing Dataplane cluster %s IDP client", cluster.ClusterID)
+	keycloakDeregistrationErr := m.osdIdpKeycloakService.DeRegisterClientInSSO(cluster.ID)
+	if keycloakDeregistrationErr != nil {
+		return errors.Wrapf(keycloakDeregistrationErr, "Failed to removed Dataplance cluster %s IDP client", cluster.ClusterID)
```

Comment:
Errors should start with lowercase letter
```suggestion
		return errors.Wrapf(keycloakDeregistrationErr, "failed to removed Dataplance cluster %s IDP client", cluster.ClusterID)
```